src/main/kotlin/ai/digital/integration/server/deploy/internals/DeployServerUtil.kt

```diff
@@ -336,35 +336,63 @@ class DeployServerUtil {
         }
 
         fun runDockerBasedInstance(project: Project, server: Server) {
-            project.exec {
-                executable = "docker-compose"
-                args = listOf("-f", getResolvedDockerFile(project, server).toFile().toString(), "up", "-d")
+            // Use ProcessBuilder instead of deprecated execOperations
+            val command = listOf("docker-compose", "-f", getResolvedDockerFile(project, server).toFile().toString(), "up", "-d")
+            val processBuilder = ProcessBuilder(command)
+
+            try {
+                val process = processBuilder.start()
+                val exitCode = process.waitFor()
+                if (exitCode != 0) {
+                    val error = process.errorStream.bufferedReader().use { it.readText() }
+                    project.logger.error("Docker compose up failed with exit code $exitCode: $error")
+                    throw RuntimeException("Docker compose up failed: $error")
+                }
+            } catch (e: Exception) {
+                project.logger.error("Failed to start docker container", e)
+                throw e
             }
         }
 
         fun stopDockerContainer(project: Project, server: Server) {
             project.logger.lifecycle("Trying to stop ${server.version} container")
-            project.exec {
-                executable = "docker-compose"
-                args = arrayListOf("-f", getResolvedDockerFile(project, server).toFile().path, "stop")
+            // Use ProcessBuilder instead of deprecated execOperations
+            val command = listOf("docker-compose", "-f", getResolvedDockerFile(project, server).toFile().path, "stop")
```

Comment:
Why isn't the **execOperations.exec** method being used in this case?

Reply:
resolved

Reply:
Updated with execOperations.exec method